test/correctness/PySys_internal_186/pysystest.py

```diff
@@ -0,0 +1,93 @@
+__pysys_title__   = r""" Process - Windows command line quoting """ 
+#                        ================================================================================
+__pysys_purpose__ = r"""
+Check that some weird edge cases when quoting Windows command lines are handled,
+and that quoted command lines are parsed the way we expect them to be.
+""" 
+	
+__pysys_authors__ = "rsm31"
+__pysys_created__ = "2022-05-30"
+#__pysys_skipped_reason__   = "Skipped until Bug-1234 is fixed"
+
+#__pysys_traceability_ids__ = "Bug-1234, UserStory-456" 
+
+import pysys
+from pysys.constants import *
+from pysys.process.helper import ProcessImpl
+
+import os, sys, math, shutil, glob
+try:
+	import win32api
+except:
+	pass
+
+class PySysTest(pysys.basetest.BaseTest):
+
+	def setup(self):
+		# We need a ProcessImpl so we can call its private methods
+		self.pimpl = ProcessImpl("", [], self.createEnvirons(), "", FOREGROUND, 0)
+
+	def testCmdLine(self, command, args, expected=None):
+		""" Test parsing of a command line built from a command + argument list
+		
+		Success means:
+		1. The quoted/escaped command line matches the expected one (if provided)
+		2. CommandLineToArgv() on the command line gets back the original arguments
+		"""
+		newcmd, cmdline = self.pimpl._ProcessImpl__buildCommandLine(command, args)
+		oldargv = [command] + args
+		newargv = win32api.CommandLineToArgv(cmdline)
+		self.log.info("-----")
+		self.log.info("Original: %s" % oldargv)
+		self.log.info("Quoted:   %s" % cmdline)
+		self.log.info("Parsed:   %s" % newargv)
+		if expected and expected != cmdline:
```

Comment:
slightly gross - why not use assertThat? This will automatically give nice (multi-line) comparison messages if there's a mismatch

Reply:
You're right, that is quite disgusting.  Fixed.